fuel-vm/src/interpreter/memory.rs

```diff
@@ -102,6 +102,13 @@ impl Memory {
         }
     }
 
+    /// Resets memory to initial state, keeping the original allocations.
+    pub fn reset(&mut self) {
+        self.stack.truncate(0);
+        self.heap.truncate(0);
```

Comment:
Truncation of the heap breaks the optimization idea. The `reverse_resize_at_least` method allocates a new vector and destroys the old one.

<img width="702" alt="image" src="https://github.com/FuelLabs/fuel-vm/assets/18346821/9290b27b-fad8-43bf-b873-dc096c84e07b">


I think we don't need to truncate the heap. We can keep it the same, but we must add logic that will zero bytes during `grow_heap`.

Reply:
Right, I forgot we're recreating the heap here. I'll fix.